### ISSUE TITLE
Nuitka ignores implicit dependencies of OpenCV for opencv-based programs.

### ISSUE THREAD

**Issue by ioctl-user** [Nuitka/Nuitka#3083]
I have custom OpenCV build, which is depends on Qt5 in linux, but doesn't include it in the whl package.
Nuitka ignores these Qt5 libs and doesn't include it in the package.

Related to the following problem: https://github.com/Nuitka/Nuitka/issues/1144

> python3 -m nuitka --version

2.4.7
Commercial: None
Python: 3.8.10 (default, Jul 29 2024, 17:02:10) 
Flavor: Debian Python
Executable: /usr/bin/python3
OS: Linux
Arch: x86_64
Distribution: Ubuntu (based on Debian) 20.04.6
Version C compiler: /usr/bin/gcc (gcc 9).

- How did you install Nuitka and Python

Python installed via package manager, nuitka installed via pip.

Full steps list to reproduce.

Run standard Ubuntu in docker container:
```
docker run -it --rm -v /tmp/out:/out ubuntu:20.04
```

Run the following commands in docker container:
```

# Prepare system
apt update && apt upgrade -y
DEBIAN_FRONTEND=noninteractive apt install -y python3-pip build-essential patchelf libgl1-mesa-glx libglib2.0-0 python3 python3-dev qt5-default git

# Build and install opencv-python with Qt dependency
git clone --recursive https://github.com/opencv/opencv-python.git

cd opencv-python/

export CMAKE_BUILD_PARALLEL_LEVEL=8
export CMAKE_ARGS="-DWITH_QT=ON"

pip wheel . --verbose
pip install opencv_python-*whl

# Generate and build simple python program with opencv
mkdir /test
cd /test

cat > 1.py << EOF
import cv2
print(cv2.__version__)
EOF

pip install nuitka
nuitka --standalone ./1.py

./1.dist/1.bin # Working

apt autoremove qt5-default # Remove system QT

./1.dist/1.bin # Not working


```

**Comment by kayhayen** [comment]
You might want to read the warnings that Nuitka gave you while it was compiling. DLLs from Debian packages are not included, only from packages not part of the OS. From what I know, opencv2 also contains a PyQt5 of its own when installed from PyPI, not sure if your wheels don't include them.

**Comment by ioctl-user** [comment]
There is no warning:

```
# nuitka --standalone  ./1.py 
Nuitka-Options: Used command line options: --standalone ./1.py
Nuitka: Starting Python compilation with Nuitka '2.4.7' on Python '3.8'
Nuitka: commercial grade 'not installed'.
Nuitka-Plugins:anti-bloat: Not including '_json' automatically in order to avoid
Nuitka-Plugins:anti-bloat: bloat, but this may cause: may slow down by using
Nuitka-Plugins:anti-bloat: fallback implementation.
Nuitka: Completed Python level compilation and optimization.
Nuitka: Generating source code for C backend compiler.
Nuitka: Running data composer tool for optimal constant value handling.         
Nuitka: Running C compilation via Scons.
Nuitka-Scons: Backend C compiler: gcc (gcc 9).
Nuitka-Scons: Backend linking program with 101 files (no progress information
Nuitka-Scons: available for this stage).
Nuitka-Scons:WARNING: You are not using ccache, re-compilation of identical code
Nuitka-Scons:WARNING: will be slower than necessary. Use your OS package manager
Nuitka-Scons:WARNING: to install it.
Nuitka: Keeping build directory '1.build'.                                      
Nuitka: Successfully created '1.dist/1.bin'.
```

Official OpenCV python build contains Qt5 libraries (directly, not through PyQt5), but one built with official site instruction is not.

Is it possible for Nuitka to check all real libraries dependencies, including system ones?

**Comment by kayhayen** [comment]
I feel that while you have no Debian package used the exclusion of dlls should be more transparent. There probably is no way to distinguish your wheel from pypi wheel or is there?

**Comment by ioctl-user** [comment]
Seems another side of this problem is Nuitka work another way with python packages, comparing to python interpreter itself.

Python looks for packages in some paths. If module present in the expected path, it's enough.
It appears, that Nuitka relies on the pip packages to work with module. So, module should have metadata to be visible by Nuitka.

The problem is some distro install python packages with package manager, and that packages doesn't contains metadata. So, pip cannot see them, python itself can, and Nuitka cannot.

**Comment by kayhayen** [comment]
> Unfortunately I cannot reproduce vanilla opencv build with custom options with dependency included.

So, there is little to do for me I guess, since I cannot support every custom built opencv myself. If you figure out the differences and how to recognize them, please make a PR.
